domain/secret/state/state.go

```diff
@@ -1707,7 +1712,7 @@ ON CONFLICT(secret_id, unit_uuid) DO UPDATE SET
 			return errors.Trace(err)
 		}
 
-		if err := st.markObsoleteRevisions(ctx, tx, uri); err != nil {
+		if err := markObsoleteRevisions(ctx, st, tx, uri); err != nil {
```

Comment:
Here's what I referred to above - we don't watch to pass in st - just use it as a receiver for the method